
package RpcEnum;

enum RpcDirection {
  UNKNOWN = 0;
  RESPONSE = 1;
  REQUEST = 2;
}

enum TeamColor {
  NEUTRAL = 0;
  BLUE = 1;
  RED = 2;
  YELLOW = 3;
}


enum RequestMethod {
  METHOD_UNSET = 0;
  PLAYER_UPDATE = 1;
  GET_PLAYER = 2;
  GET_INVENTORY = 4;
  DOWNLOAD_SETTINGS = 5;
  DOWNLOAD_ITEM_TEMPLATES = 6;
  DOWNLOAD_REMOTE_CONFIG_VERSION = 7;
  FORT_SEARCH = 101;
  ENCOUNTER = 102;
  CATCH_POKEMON = 103;
  FORT_DETAILS = 104;
  ITEM_USE = 105;
  GET_MAP_OBJECTS = 106;
  FORT_DEPLOY_POKEMON = 110;
  FORT_RECALL_POKEMON = 111;
  RELEASE_POKEMON = 112;
  USE_ITEM_POTION = 113;
  USE_ITEM_CAPTURE = 114;
  USE_ITEM_FLEE = 115;
  USE_ITEM_REVIVE = 116;
  TRADE_SEARCH = 117;
  TRADE_OFFER = 118;
  TRADE_RESPONSE = 119;
  TRADE_RESULT = 120;
  GET_PLAYER_PROFILE = 121;
  GET_ITEM_PACK = 122;
  BUY_ITEM_PACK = 123;
  BUY_GEM_PACK = 124;
  EVOLVE_POKEMON = 125;
  GET_HATCHED_EGGS = 126;
  ENCOUNTER_TUTORIAL_COMPLETE = 127;
  LEVEL_UP_REWARDS = 128;
  CHECK_AWARDED_BADGES = 129;
  USE_ITEM_GYM = 133;
  GET_GYM_DETAILS = 134;
  START_GYM_BATTLE = 135;
  ATTACK_GYM = 136;
  RECYCLE_INVENTORY_ITEM = 137;
  COLLECT_DAILY_BONUS = 138;
  USE_ITEM_XP_BOOST = 139;
  USE_ITEM_EGG_INCUBATOR = 140;
  USE_INCENSE = 141;
  GET_INCENSE_POKEMON = 142;
  INCENSE_ENCOUNTER = 143;
  ADD_FORT_MODIFIER = 144;
  DISK_ENCOUNTER = 145;
  COLLECT_DAILY_DEFENDER_BONUS = 146;
  UPGRADE_POKEMON = 147;
  SET_FAVORITE_POKEMON = 148;
  NICKNAME_POKEMON = 149;
  EQUIP_BADGE = 150;
  SET_CONTACT_SETTINGS = 151;
  GET_ASSET_DIGEST = 300;
  GET_DOWNLOAD_URLS = 301;
  GET_SUGGESTED_CODENAMES = 401;
  CHECK_CODENAME_AVAILABLE = 402;
  CLAIM_CODENAME = 403;
  SET_AVATAR = 404;
  SET_PLAYER_TEAM = 405;
  MARK_TUTORIAL_COMPLETE = 406;
  LOAD_SPAWN_POINTS = 500;
  ECHO = 666;
  DEBUG_UPDATE_INVENTORY = 700;
  DEBUG_DELETE_PLAYER = 701;
  SFIDA_REGISTRATION = 800;
  SFIDA_ACTION_LOG = 801;
  SFIDA_CERTIFICATION = 802;
  SFIDA_UPDATE = 803;
  SFIDA_ACTION = 804;
  SFIDA_DOWSER = 805;
  SFIDA_CAPTURE = 806;
}

enum PokemonMove {
  MOVE_UNSET = 0;
  THUNDER_SHOCK = 1;
  QUICK_ATTACK = 2;
  SCRATCH = 3;
  EMBER = 4;
  VINE_WHIP = 5;
  TACKLE = 6;
  RAZOR_LEAF = 7;
  TAKE_DOWN = 8;
  WATER_GUN = 9;
  BITE = 10;
  POUND = 11;
  DOUBLE_SLAP = 12;
  WRAP = 13;
  HYPER_BEAM = 14;
  LICK = 15;
  DARK_PULSE = 16;
  SMOG = 17;
  SLUDGE = 18;
  METAL_CLAW = 19;
  VICE_GRIP = 20;
  FLAME_WHEEL = 21;
  MEGAHORN = 22;
  WING_ATTACK = 23;
  FLAMETHROWER = 24;
  SUCKER_PUNCH = 25;
  DIG = 26;
  LOW_KICK = 27;
  CROSS_CHOP = 28;
  PSYCHO_CUT = 29;
  PSYBEAM = 30;
  EARTHQUAKE = 31;
  STONE_EDGE = 32;
  ICE_PUNCH = 33;
  HEART_STAMP = 34;
  DISCHARGE = 35;
  FLASH_CANNON = 36;
  PECK = 37;
  DRILL_PECK = 38;
  ICE_BEAM = 39;
  BLIZZARD = 40;
  AIR_SLASH = 41;
  HEAT_WAVE = 42;
  TWINEEDLE = 43;
  POISON_JAB = 44;
  AERIAL_ACE = 45;
  DRILL_RUN = 46;
  PETAL_BLIZZARD = 47;
  MEGA_DRAIN = 48;
  BUG_BUZZ = 49;
  POISON_FANG = 50;
  NIGHT_SLASH = 51;
  SLASH = 52;
  BUBBLE_BEAM = 53;
  SUBMISSION = 54;
  KARATE_CHOP = 55;
  LOW_SWEEP = 56;
  AQUA_JET = 57;
  AQUA_TAIL = 58;
  SEED_BOMB = 59;
  PSYSHOCK = 60;
  ROCK_THROW = 61;
  ANCIENT_POWER = 62;
  ROCK_TOMB = 63;
  ROCK_SLIDE = 64;
  POWER_GEM = 65;
  SHADOW_SNEAK = 66;
  SHADOW_PUNCH = 67;
  SHADOW_CLAW = 68;
  OMINOUS_WIND = 69;
  SHADOW_BALL = 70;
  BULLET_PUNCH = 71;
  MAGNET_BOMB = 72;
  STEEL_WING = 73;
  IRON_HEAD = 74;
  PARABOLIC_CHARGE = 75;
  SPARK = 76;
  THUNDER_PUNCH = 77;
  THUNDER = 78;
  THUNDERBOLT = 79;
  TWISTER = 80;
  DRAGON_BREATH = 81;
  DRAGON_PULSE = 82;
  DRAGON_CLAW = 83;
  DISARMING_VOICE = 84;
  DRAINING_KISS = 85;
  DAZZLING_GLEAM = 86;
  MOONBLAST = 87;
  PLAY_ROUGH = 88;
  CROSS_POISON = 89;
  SLUDGE_BOMB = 90;
  SLUDGE_WAVE = 91;
  GUNK_SHOT = 92;
  MUD_SHOT = 93;
  BONE_CLUB = 94;
  BULLDOZE = 95;
  MUD_BOMB = 96;
  FURY_CUTTER = 97;
  BUG_BITE = 98;
  SIGNAL_BEAM = 99;
  X_SCISSOR = 100;
  FLAME_CHARGE = 101;
  FLAME_BURST = 102;
  FIRE_BLAST = 103;
  BRINE = 104;
  WATER_PULSE = 105;
  SCALD = 106;
  HYDRO_PUMP = 107;
  PSYCHIC = 108;
  PSYSTRIKE = 109;
  ICE_SHARD = 110;
  ICY_WIND = 111;
  FROST_BREATH = 112;
  ABSORB = 113;
  GIGA_DRAIN = 114;
  FIRE_PUNCH = 115;
  SOLAR_BEAM = 116;
  LEAF_BLADE = 117;
  POWER_WHIP = 118;
  SPLASH = 119;
  ACID = 120;
  AIR_CUTTER = 121;
  HURRICANE = 122;
  BRICK_BREAK = 123;
  CUT = 124;
  SWIFT = 125;
  HORN_ATTACK = 126;
  STOMP = 127;
  HEADBUTT = 128;
  HYPER_FANG = 129;
  SLAM = 130;
  BODY_SLAM = 131;
  REST = 132;
  STRUGGLE = 133;
  SCALD_BLASTOISE = 134;
  HYDRO_PUMP_BLASTOISE = 135;
  WRAP_GREEN = 136;
  WRAP_PINK = 137;
  FURY_CUTTER_FAST = 200;
  BUG_BITE_FAST = 201;
  BITE_FAST = 202;
  SUCKER_PUNCH_FAST = 203;
  DRAGON_BREATH_FAST = 204;
  THUNDER_SHOCK_FAST = 205;
  SPARK_FAST = 206;
  LOW_KICK_FAST = 207;
  KARATE_CHOP_FAST = 208;
  EMBER_FAST = 209;
  WING_ATTACK_FAST = 210;
  PECK_FAST = 211;
  LICK_FAST = 212;
  SHADOW_CLAW_FAST = 213;
  VINE_WHIP_FAST = 214;
  RAZOR_LEAF_FAST = 215;
  MUD_SHOT_FAST = 216;
  ICE_SHARD_FAST = 217;
  FROST_BREATH_FAST = 218;
  QUICK_ATTACK_FAST = 219;
  SCRATCH_FAST = 220;
  TACKLE_FAST = 221;
  POUND_FAST = 222;
  CUT_FAST = 223;
  POISON_JAB_FAST = 224;
  ACID_FAST = 225;
  PSYCHO_CUT_FAST = 226;
  ROCK_THROW_FAST = 227;
  METAL_CLAW_FAST = 228;
  BULLET_PUNCH_FAST = 229;
  WATER_GUN_FAST = 230;
  SPLASH_FAST = 231;
  WATER_GUN_FAST_BLASTOISE = 232;
  MUD_SLAP_FAST = 233;
  ZEN_HEADBUTT_FAST = 234;
  CONFUSION_FAST = 235;
  POISON_STING_FAST = 236;
  BUBBLE_FAST = 237;
  FEINT_ATTACK_FAST = 238;
  STEEL_WING_FAST = 239;
  FIRE_FANG_FAST = 240;
  ROCK_SMASH_FAST = 241;
}


enum ItemType {
  ITEM_UNKNOWN = 0;
  ITEM_POKE_BALL = 1;
  ITEM_GREAT_BALL = 2;
  ITEM_ULTRA_BALL = 3;
  ITEM_MASTER_BALL = 4;
  ITEM_POTION = 101;
  ITEM_SUPER_POTION = 102;
  ITEM_HYPER_POTION = 103;
  ITEM_MAX_POTION = 104;
  ITEM_REVIVE = 201;
  ITEM_MAX_REVIVE = 202;
  ITEM_LUCKY_EGG = 301;
  ITEM_INCENSE_ORDINARY = 401;
  ITEM_INCENSE_SPICY = 402;
  ITEM_INCENSE_COOL = 403;
  ITEM_INCENSE_FLORAL = 404;
  ITEM_TROY_DISK = 501;
  ITEM_X_ATTACK = 602;
  ITEM_X_DEFENSE = 603;
  ITEM_X_MIRACLE = 604;
  ITEM_RAZZ_BERRY = 701;
  ITEM_BLUK_BERRY = 702;
  ITEM_NANAB_BERRY = 703;
  ITEM_WEPAR_BERRY = 704;
  ITEM_PINAP_BERRY = 705;
  ITEM_SPECIAL_CAMERA = 801;
  ITEM_INCUBATOR_BASIC_UNLIMITED = 901;
  ITEM_INCUBATOR_BASIC = 902;
  ITEM_POKEMON_STORAGE_UPGRADE = 1001;
  ITEM_ITEM_STORAGE_UPGRADE = 1002;
}

enum InventoryUpgradeType {
  UPGRADE_UNSET = 0;
  INCREASE_ITEM_STORAGE = 1;
  INCREASE_POKEMON_STORAGE = 2;
}

enum ItemTypeCategory {
  ITEM_TYPE_NONE = 0;
  ITEM_TYPE_POKEBALL = 1;
  ITEM_TYPE_POTION = 2;
  ITEM_TYPE_REVIVE = 3;
  ITEM_TYPE_MAP = 4;
  ITEM_TYPE_BATTLE = 5;
  ITEM_TYPE_FOOD = 6;
  ITEM_TYPE_CAMERA = 7;
  ITEM_TYPE_DISK = 8;
  ITEM_TYPE_INCUBATOR = 9;
  ITEM_TYPE_INCENSE = 10;
  ITEM_TYPE_XP_BOOST = 11;
  ITEM_TYPE_INVENTORY_UPGRADE = 12;
}

enum EggIncubatorType {
  INCUBATOR_UNSET = 0;
  INCUBATOR_DISTANCE = 1;
}

enum PokemonFamilyId {
  FAMILY_UNSET = 0;
  FAMILY_BULBASAUR = 1;
  FAMILY_CHARMANDER = 4;
  FAMILY_SQUIRTLE = 7;
  FAMILY_CATERPIE = 10;
  FAMILY_WEEDLE = 13;
  FAMILY_PIDGEY = 16;
  FAMILY_RATTATA = 19;
  FAMILY_SPEAROW = 21;
  FAMILY_EKANS = 23;
  FAMILY_PIKACHU = 25;
  FAMILY_SANDSHREW = 27;
  FAMILY_NIDORAN = 29;
  FAMILY_NIDORAN2 = 32;
  FAMILY_CLEFAIRY = 35;
  FAMILY_VULPIX = 37;
  FAMILY_JIGGLYPUFF = 39;
  FAMILY_ZUBAT = 41;
  FAMILY_ODDISH = 43;
  FAMILY_PARAS = 46;
  FAMILY_VENONAT = 48;
  FAMILY_DIGLETT = 50;
  FAMILY_MEOWTH = 52;
  FAMILY_PSYDUCK = 54;
  FAMILY_MANKEY = 56;
  FAMILY_GROWLITHE = 58;
  FAMILY_POLIWAG = 60;
  FAMILY_ABRA = 63;
  FAMILY_MACHOP = 66;
  FAMILY_BELLSPROUT = 69;
  FAMILY_TENTACOOL = 72;
  FAMILY_GEODUDE = 74;
  FAMILY_PONYTA = 77;
  FAMILY_SLOWPOKE = 79;
  FAMILY_MAGNEMITE = 81;
  FAMILY_FARFETCHD = 83;
  FAMILY_DODUO = 84;
  FAMILY_SEEL = 86;
  FAMILY_GRIMER = 88;
  FAMILY_SHELLDER = 90;
  FAMILY_GASTLY = 92;
  FAMILY_ONIX = 95;
  FAMILY_DROWZEE = 96;
  FAMILY_KRABBY = 98;
  FAMILY_VOLTORB = 100;
  FAMILY_EXEGGCUTE = 102;
  FAMILY_CUBONE = 104;
  FAMILY_HITMONLEE = 106;
  FAMILY_HITMONCHAN = 107;
  FAMILY_LICKITUNG = 108;
  FAMILY_KOFFING = 109;
  FAMILY_RHYHORN = 111;
  FAMILY_CHANSEY = 113;
  FAMILY_TANGELA = 114;
  FAMILY_KANGASKHAN = 115;
  FAMILY_HORSEA = 116;
  FAMILY_GOLDEEN = 118;
  FAMILY_STARYU = 120;
  FAMILY_MR_MIME = 122;
  FAMILY_SCYTHER = 123;
  FAMILY_JYNX = 124;
  FAMILY_ELECTABUZZ = 125;
  FAMILY_MAGMAR = 126;
  FAMILY_PINSIR = 127;
  FAMILY_TAUROS = 128;
  FAMILY_MAGIKARP = 129;
  FAMILY_LAPRAS = 131;
  FAMILY_DITTO = 132;
  FAMILY_EEVEE = 133;
  FAMILY_PORYGON = 137;
  FAMILY_OMANYTE = 138;
  FAMILY_KABUTO = 140;
  FAMILY_AERODACTYL = 142;
  FAMILY_SNORLAX = 143;
  FAMILY_ARTICUNO = 144;
  FAMILY_ZAPDOS = 145;
  FAMILY_MOLTRES = 146;
  FAMILY_DRATINI = 147;
  FAMILY_MEWTWO = 150;
  FAMILY_MEW = 151;
}

enum MapObjectsStatus {
  UNSET_STATUS = 0;
  SUCCESS = 1;
  LOCATION_UNSET = 2;
}

enum FortType {
  GYM = 0;
  CHECKPOINT = 1;
}

enum PokemonId {
  MISSINGNO = 0;
  BULBASAUR = 1;
  IVYSAUR = 2;
  VENUSAUR = 3;
  CHARMENDER = 4;
  CHARMELEON = 5;
  CHARIZARD = 6;
  SQUIRTLE = 7;
  WARTORTLE = 8;
  BLASTOISE = 9;
  CATERPIE = 10;
  METAPOD = 11;
  BUTTERFREE = 12;
  WEEDLE = 13;
  KAKUNA = 14;
  BEEDRILL = 15;
  PIDGEY = 16;
  PIDGEOTTO = 17;
  PIDGEOT = 18;
  RATTATA = 19;
  RATICATE = 20;
  SPEAROW = 21;
  FEAROW = 22;
  EKANS = 23;
  ARBOK = 24;
  PIKACHU = 25;
  RAICHU = 26;
  SANDSHREW = 27;
  SANDLASH = 28;
  NIDORAN_FEMALE = 29;
  NIDORINA = 30;
  NIDOQUEEN = 31;
  NIDORAN_MALE = 32;
  NIDORINO = 33;
  NIDOKING = 34;
  CLEFARY = 35;
  CLEFABLE = 36;
  VULPIX = 37;
  NINETALES = 38;
  JIGGLYPUFF = 39;
  WIGGLYTUFF = 40;
  ZUBAT = 41;
  GOLBAT = 42;
  ODDISH = 43;
  GLOOM = 44;
  VILEPLUME = 45;
  PARAS = 46;
  PARASECT = 47;
  VENONAT = 48;
  VENOMOTH = 49;
  DIGLETT = 50;
  DUGTRIO = 51;
  MEOWTH = 52;
  PERSIAN = 53;
  PSYDUCK = 54;
  GOLDUCK = 55;
  MANKEY = 56;
  PRIMEAPE = 57;
  GROWLITHE = 58;
  ARCANINE = 59;
  POLIWAG = 60;
  POLIWHIRL = 61;
  POLIWRATH = 62;
  ABRA = 63;
  KADABRA = 64;
  ALAKHAZAM = 65;
  MACHOP = 66;
  MACHOKE = 67;
  MACHAMP = 68;
  BELLSPROUT = 69;
  WEEPINBELL = 70;
  VICTREEBELL = 71;
  TENTACOOL = 72;
  TENTACRUEL = 73;
  GEODUGE = 74;
  GRAVELER = 75;
  GOLEM = 76;
  PONYTA = 77;
  RAPIDASH = 78;
  SLOWPOKE = 79;
  SLOWBRO = 80;
  MAGNEMITE = 81;
  MAGNETON = 82;
  FARFETCHD = 83;
  DODUO = 84;
  DODRIO = 85;
  SEEL = 86;
  DEWGONG = 87;
  GRIMER = 88;
  MUK = 89;
  SHELLDER = 90;
  CLOYSTER = 91;
  GASTLY = 92;
  HAUNTER = 93;
  GENGAR = 94;
  ONIX = 95;
  DROWZEE = 96;
  HYPNO = 97;
  KRABBY = 98;
  KINGLER = 99;
  VOLTORB = 100;
  ELECTRODE = 101;
  EXEGGCUTE = 102;
  EXEGGUTOR = 103;
  CUBONE = 104;
  MAROWAK = 105;
  HITMONLEE = 106;
  HITMONCHAN = 107;
  LICKITUNG = 108;
  KOFFING = 109;
  WEEZING = 110;
  RHYHORN = 111;
  RHYDON = 112;
  CHANSEY = 113;
  TANGELA = 114;
  KANGASKHAN = 115;
  HORSEA = 116;
  SEADRA = 117;
  GOLDEEN = 118;
  SEAKING = 119;
  STARYU = 120;
  STARMIE = 121;
  MR_MIME = 122;
  SCYTHER = 123;
  JYNX = 124;
  ELECTABUZZ = 125;
  MAGMAR = 126;
  PINSIR = 127;
  TAUROS = 128;
  MAGIKARP = 129;
  GYARADOS = 130;
  LAPRAS = 131;
  DITTO = 132;
  EEVEE = 133;
  VAPOREON = 134;
  JOLTEON = 135;
  FLAREON = 136;
  PORYGON = 137;
  OMANYTE = 138;
  OMASTAR = 139;
  KABUTO = 140;
  KABUTOPS = 141;
  AERODACTYL = 142;
  SNORLAX = 143;
  ARTICUNO = 144;
  ZAPDOS = 145;
  MOLTRES = 146;
  DRATINI = 147;
  DRAGONAIR = 148;
  DRAGONITE = 149;
  MEWTWO = 150;
  MEW = 151;
}

enum FortSponsor {
  UNSET_SPONSOR = 0;
  MCDONALDS = 1;
  POKEMON_STORE = 2;
}

enum FortRenderingType {
  DEFAULT = 0;
  INTERNAL_TEST = 1;
}